src/main/java/org/opentripplanner/middleware/utils/ConfigUtils.java

```diff
@@ -53,6 +58,38 @@ public static void loadConfig(String[] args) throws IOException {
             envConfigStream = new FileInputStream(new File(args[0]));
         }
         envConfig = yamlMapper.readTree(envConfigStream);
+        validateConfig();
+    }
+
+    /**
+     * Validate the environment configuration against the environment configuration schema. If the config is not available,
+     * does not match the schema or an exception is thrown, exit the application.
+     */
+    private static void validateConfig() {
+        if ("false".equals(getConfigPropertyAsText("VALIDATE_ENVIRONMENT_CONFIG", "true"))) {
+            LOG.warn("Environment configuration schema validation disabled.");
+            return;
+        }
+        try {
+            if (envConfig == null) {
+                throw new IllegalArgumentException("Environment configuration not available to validate!");
+            }
+            FileInputStream envSchemaStream = new FileInputStream(new File(DEFAULT_ENV_SCHEMA));
```

Comment:
It looks like here (and in some other spots in this file), creating the FileInputStream with a File arg is redundant (I think we can just pass a String).

Reply:
Updated to just pass a String.